### ISSUE TITLE
"[ERROR] Network is down, exiting this Docker" - can't find further info in logs

### ISSUE THREAD

**Issue by FunctionDJ** [DyonR/docker-qbittorrentvpn#90]
Hi, i'm running this container using portainer and tried to follow the README as closely as possible, but i'm getting a "Network is down" error, which exits the container and makes it 'boot-loop' because of the restart policy.

Here's an edited screenshot of my config in portainer:

![image](https://user-images.githubusercontent.com/34871211/153755713-8b594d16-f163-4fe5-abcd-810e06f4a303.png)

I'm also running the container in privileged mode because i ran into [this issue](#39).

Finally, here's my edited log file that includes one full "run" of the container before it loops:

[_qbittorrent-with-vpn_logs.txt](https://github.com/DyonR/docker-qbittorrentvpn/files/8055625/_qbittorrent-with-vpn_logs.txt)

I'm very thankful for any help :pray:

**Comment by DyonR** [comment]
I have added a new environment variable called `RESTART_CONTAINER` if you set this to `no` it will do nothing when the "network is down".  
Please update the container, set `RESTART_CONTAINER` to `no` and check if the container can access the internet.  
You can do that by running the following commands from the console:  
`apt update && apt install -y curl dnsutils`
`curl ifconfig.io` > If this shows your vpn ip you are connected to your vpn provider  
After that you can perhaps troubleshoot a bit why it can't connect to the internet.  
Try running some of the following commands;  
`nslookup one.one.one.one`  
`nslookup github.com`
`ping one.one.one.one`
`ping github.com`
`cat /etc/resolv.conf` > To check which DNS servers the Docker uses (should also show while running nslookup actually)

**Comment by FunctionDJ** [comment]
@DyonR `curl ifconfig.io` gives me the correct VPN IP address. all of the other commands also work, i can reach cloudflare's DNS and github all fine. the resolv.conf looks like this:
```
# Some comment stuff

nameserver 192.168.0.1
search myrouter.name
nameserver 1.1.1.1
nameserver 1.0.0.1
```
does this info help? logs still say `[ERROR] Network is possibly down`

**Comment by pmd5700** [comment]
Have you been able to sort anything out with this? I get the similar `[ERROR] Network is possibly down`. However, if I have `RESTART_CONTAINER` on or off it still just stops and doesn't restart. 

I currently have `RESTART_CONTAINER` set to `yes`, and the log shows that it tries to restart container, but it doesn't succeed as far as I can tell.

<html>
<body>


date | stream | content
-- | -- | --
2022-03-15 00:30:10 | stdout | 2022-03-14 20:30:10.707405 [INFO] Restarting container.
2022-03-15 00:30:09 | stdout | 2022-03-14 20:30:09.599614 [ERROR] Network is possibly down.
2022-03-15 00:27:42 | stdout | 2022-03-14 20:27:42 TLS Error: local/remote TLS keys are out of sync: [AF_INET]IP REMOVED [0]
2022-03-15 00:24:17 | stdout | 2022-03-14 20:24:17 TLS Error: local/remote TLS keys are out of sync: [AF_INET]IP REMOVED [0]
2022-03-15 00:23:08 | stdout | 2022-03-14 20:23:08 TLS Error: local/remote TLS keys are out of sync: [AF_INET]IP REMOVED [0]
2022-03-15 00:03:35 | stdout | 2022-03-14 20:03:35 Control Channel: TLSv1.3, cipher TLSv1.3 TLS_AES_256_GCM_SHA384, 4096 bit RSA
2022-03-15 00:03:35 | stdout | 2022-03-14 20:03:35 Incoming Data Channel: Cipher 'AES-256-GCM' initialized with 256 bit key
2022-03-15 00:03:35 | stdout | 2022-03-14 20:03:35 Outgoing Data Channel: Cipher 'AES-256-GCM' initialized with 256 bit key
2022-03-15 00:03:35 | stdout | 2022-03-14 20:03:35 VERIFY OK: depth=0, C=CA, ST=ON, L=Toronto, O=Windscribe Limited, OU=Systems, CN=mia-140.windscribe.com
2022-03-15 00:03:35 | stdout | 2022-03-14 20:03:35 VERIFY X509NAME OK: C=CA, ST=ON, L=Toronto, O=Windscribe Limited, OU=Systems, CN=mia-140.windscribe.com
2022-03-15 00:03:35 | stdout | 2022-03-14 20:03:35 VERIFY EKU OK
2022-03-15 00:03:35 | stdout | 2022-03-14 20:03:35 ++ Certificate has EKU (str) TLS Web Server Authentication, expects TLS Web Server Authentication
2022-03-15 00:03:35 | stdout | 2022-03-14 20:03:35 Validating certificate extended key usage
2022-03-15 00:03:35 | stdout | 2022-03-14 20:03:35 VERIFY KU OK
2022-03-15 00:03:35 | stdout | 2022-03-14 20:03:35 VERIFY OK: depth=1, C=CA, ST=ON, L=Toronto, O=Windscribe Limited, OU=Systems, CN=Windscribe Node CA X2
2022-03-15 00:03:35 | stdout | 2022-03-14 20:03:35 VERIFY OK: depth=2, C=CA, ST=ON, L=Toronto, O=Windscribe Limited, OU=Systems, CN=Windscribe Node CA X1


</body>
</html>

**Comment by Pondiniii** [comment]
hello,
I am using windscribe VPN. Windscribe-Wireguard didn't work for me for some reason, but Windscribe-OpenVPN is working fine.

**Comment by pmd5700** [comment]
This is happening multiple times per day now. Is there any way to force a container to restart in the Synolgy Software?

**Comment by WhiteNight121** [comment]
@FunctionDJ  i just had this issue but i ended up just restarting the docker service guessing / hoping it was a one time thing. ether way did you test the openvpn on another device to make sure it works?
@pmd5700  do you have the 'latest' image set?

**Comment by pmd5700** [comment]
@WhiteNight121 Yes, I have latest set. I thought it was getting better - didn't have an issue for a week or so. Now it's back to every day or 2.

**Comment by WhiteNight121** [comment]
> @WhiteNight121 Yes, I have latest set. I thought it was getting better - didn't have an issue for a week or so. Now it's back to every day or 2.

Hmm I do see the VM go down and restart or I assume it does as if I have the tab open sometimes I got log back in again but it comes back up and goes back to doing it's thing. I will say I'm not on the latest
